### PR TITLE
Make sure we don't update title twice

### PR DESCRIPTION
**Pull request by Nyholm**
I want to avoid this: 

Screenshot 2020-11-12 at 22 38 44

----------
--------
--------
--------

It happens when a user adds two labels at the same time. 

FYI @jderusse

### CODE REVIEW

File: src/Subscriber/AutoUpdateTitleWithLabelSubscriber.php [hunk 1]
```diff
@@ -69,10 +72,22 @@ public function onPullRequest(GitHubEvent $event)
...
+        $lock = $this->lockFactory->createLock($repository->getFullName().'#'.$number);
+        $lock->acquire(true); // blocking
+        try {
+            // Refetch the current title just to make sure it has not changed
+            if ($prTitle === ($this->pullRequestApi->show($repository, $number)['title'] ?? '')) {
+                return;
+            }
+            $this->pullRequestApi->updateTitle($repository, $number, $prTitle);
+        } finally {
+            $lock->release();
```
Comment (jderusse) [inline src/Subscriber/AutoUpdateTitleWithLabelSubscriber.php:86]: not needed, Symfony do it for you on `__destruct`

File: config/packages/lock.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+services:
+
+    app.lock.store:
+        class: Symfony\Component\Lock\Store\PdoStore
+        factory: [Symfony\Component\Lock\Store\StoreFactory,  createStore]
+        arguments: ['%env(DATABASE_URL)%']
+
+    app.lock.blocking_store:
```
Comment (jderusse) [inline config/packages/lock.yaml:8]: not needed with symfony/lock 5.2

File: src/Subscriber/AutoUpdateTitleWithLabelSubscriber.php (hunk 1 above)
Comment (Nyholm) [reply src/Subscriber/AutoUpdateTitleWithLabelSubscriber.php:86]: Oh, Thank you

File: config/packages/lock.yaml (hunk 2 above)
Comment (Nyholm) [reply config/packages/lock.yaml:8]: I'll have to update then =)

Thank you

**Comment by Nyholm** [comment]
Thank you for the review. The PR is updated

File: composer.json [hunk 3]
```diff
@@ -68,5 +69,6 @@
             "assets:install --symlink --relative %PUBLIC_DIR%": "symfony-cmd",
             "assets:install %PUBLIC_DIR%": "symfony-cmd"
         }
-    }
+    },
+    "minimum-stability": "RC"
```
Comment (jderusse) [inline composer.json:73]: I would also adds `prefer-stable: true`

File: symfony.lock [hunk 4]
```diff
@@ -268,6 +268,9 @@
     "symfony/http-kernel": {
         "version": "v5.1.2"
     },
+    "symfony/lock": {
+        "version": "v5.1.8"
```
Comment (jderusse) [inline symfony.lock:272]: Shouldn't it be 5.2?

**Review by jderusse — APPROVED** [review]
Thank you @Nyholm
A nitpicking comment from my side :)

File: composer.json (hunk 3 above)
Comment (Nyholm) [reply composer.json:73]: I ran composer update with that command option.

File: symfony.lock (hunk 4 above)
Comment (Nyholm) [reply symfony.lock:272]: I am not sure. I did not modify this myself. 

I dont think it matters too much.

**Review by wouterj — APPROVED** [review]

**Pull request merged**